pkg/virt-config/config-map.go

```diff
@@ -99,18 +102,81 @@ func getConfigMap() *k8sv1.ConfigMap {
 // 1. Check if the config exists. If it does not exist, return the default config
 // 2. Check if the config got updated. If so, try to parse and return it
 // 3. In case of errors or no updates (resource version stays the same), it returns the values from the last good config
-func NewClusterConfig(configMapInformer cache.SharedIndexInformer, namespace string) *ClusterConfig {
+func NewClusterConfig(configMapInformer cache.SharedIndexInformer, crdInformer cache.SharedIndexInformer, namespace string) *ClusterConfig {
 
 	c := &ClusterConfig{
-		informer:        configMapInformer,
-		lock:            &sync.Mutex{},
-		namespace:       namespace,
-		lastValidConfig: defaultClusterConfig(),
-		defaultConfig:   defaultClusterConfig(),
+		configMapInformer: configMapInformer,
+		crdInformer:       crdInformer,
+		lock:              &sync.Mutex{},
+		namespace:         namespace,
+		lastValidConfig:   defaultClusterConfig(),
+		defaultConfig:     defaultClusterConfig(),
 	}
+
+	c.configMapInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.configAddedDeleted,
+		DeleteFunc: c.configAddedDeleted,
+		UpdateFunc: c.configUpdated,
+	})
+
+	c.crdInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.crdAddedDeleted,
+		DeleteFunc: c.crdAddedDeleted,
+		UpdateFunc: c.crdUpdated,
+	})
+
 	return c
 }
 
+func (c *ClusterConfig) configAddedDeleted(obj interface{}) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	if c.configModifiedCallback != nil {
+		go c.configModifiedCallback()
+	}
+}
+func (c *ClusterConfig) configUpdated(old, cur interface{}) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	if c.configModifiedCallback != nil {
+		go c.configModifiedCallback()
+	}
+}
+
+func isDataVolumeCrd(crd *extv1beta1.CustomResourceDefinition) bool {
+	if crd.Spec.Names.Kind == "DataVolume" {
+		return true
+	}
+
+	return false
+
+}
+
+func (c *ClusterConfig) crdAddedDeleted(obj interface{}) {
```

Comment:
what does this function do? sorry, i don't get the name

Reply:
informer callback.  it's called whenever a crd is added or deleted